src/components/saveChangesModal/index.js

```diff
@@ -0,0 +1,89 @@
+import { useNavigate } from "react-router-dom"
+import { useState } from "react";
+import useModal from "../../hooks/useModal"
+import './style.css'
+import Button from '../button'
+import Toast from '../toast'
+import { patch } from "../../service/apiClient";
+import ErrorMessage from "../errorMessage";
+
+const SaveChangesModal = ({ formState, id, loggedInUserInfo }) => {
+    const [isError, setIsError] = useState(false)
+    const { closeModal } = useModal()
+    const navigate = useNavigate()
+    const handleDontSave = () => {       
+            closeModal()
+            navigate(`/profile/${id}`)
+    }
+
+    const handleCancel = () => {
+        closeModal()
+        navigate(`/profile/${id}/edit`)
+    }
+
+    const handleSave = async () => {
+        await handleSubmit()
+        if(isError === false){
+            closeModal()
+            navigate(`/profile/${id}`)
+        }
+        if(isError === true){
+            console.log("there is an error")
+        }
+      
+    }
+    const handleSubmit = async () => {
+        const editedProfile = formState
+        const formDataPATCH = async () => {         
```

Comment:
This function is not needed since you have defined `handleSubmit` as `async` already.